### PR TITLE
Markdown commit for test smarter blog Lierheimer/Kumalah

### PR DESCRIPTION
**Pull request by faithebear**
Nov 2024

## What are you changing in this pull request and why?

This blog is part of a training team Q3 initiative to bring training field experience to the developer blog. See[ notion task ](https://www.notion.so/dbtlabs/Faith-Jerrie-Q3-Dev-Blog-Post-9144a6f9a7f24ad68507c817da8bb531?pvs=4 )for more detail.

## Checklist
- [x] I have reviewed the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) so my content adheres to these guidelines.
- [x] The topic I'm writing about is for specific dbt version(s) and I have versioned it according to the [version a whole page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) and/or [version a block of content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content) guidelines.
- [x] I have added checklist item(s) to this list for anything anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."
- [x] The content in this PR requires a dbt release note, so I added one to the [release notes page](https://docs.getdbt.com/docs/dbt-versions/dbt-cloud-release-notes).

### CODE REVIEW

File: website/blog/2024-11-04-test-smarter-not-harder.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+- identifying *data hygiene* issues
+- identifying *business-focused anomaly* issues
+- identifying *stats-focused anomaly* issues
+
+Once we have defined data quality*,* we’ll move on to *prioritize* those concerns. We will:
+
+- think through each concern in terms of the breadth of impact
+- decide if each concern should be at error or warning severity
+
```
Comment (joellabes) [inline website/blog/2024-11-04-test-smarter-not-harder.md:29]:
```suggestion

<!-- truncate -->

```

Comment (joellabes) [reply website/blog/2024-11-04-test-smarter-not-harder.md:29]: This ends the preview section in the [blog homepage](https://docs-getdbt-com-git-lierheimer-test-smarter-blo-273af8-dbt-labs.vercel.app/blog)

File: website/blog/2024-11-04-test-smarter-not-harder.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+The [Analytics Development Lifecycle (ADLC)](https://www.getdbt.com/resources/guides/the-analytics-development-lifecycle) is a workflow for improving data maturity and velocity. Testing is a key phase here. Many dbt developers tend to focus on [primary keys and source freshness.](https://www.getdbt.com/blog/building-a-data-quality-framework-with-dbt-and-dbt-cloud) We think there is a more holistic and in-depth path to tread. Testing is a key piece of the ADLC, and it should drive data quality.
+
+In this blog, we’ll walk through a plan to define data quality. This will look like:
+
+- identifying *data hygiene* issues
+- identifying *business-focused anomaly* issues
+- identifying *stats-focused anomaly* issues
+
+Once we have defined data quality*,* we’ll move on to *prioritize* those concerns. We will:
```
Comment (joellabes) [inline website/blog/2024-11-04-test-smarter-not-harder.md]:
```suggestion
Once we have *defined* data quality, we’ll move on to *prioritize* those concerns. We will:
```
(I assume)

File: website/blog/2024-11-04-test-smarter-not-harder.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+## Identifying data quality issues in your pipeline
+
+Let’s start our framework by *identifying* types of data quality issues.
+
+In our daily work with customers, we find that data quality issues tend to fall into one of three broad buckets: *data hygiene, business-focused anomalies,* and *stats-focused anomalies.* Read the bucket descriptions below, and list 2-3 data quality concerns in your own business context that fall into each bucket.
+
+### Bucket One: Data Hygiene
+
+*Data hygiene* issues **are concerns you address in your [staging layer.](https://docs.getdbt.com/best-practices/how-we-structure/2-staging) Hygienic data meets your expectations around formatting, completeness, and granularity requirements. Here are a few examples.
```
Comment (joellabes) [inline website/blog/2024-11-04-test-smarter-not-harder.md]:
```suggestion
*Data hygiene* issues are concerns you address in your [staging layer.](https://docs.getdbt.com/best-practices/how-we-structure/2-staging) Hygienic data meets your expectations around formatting, completeness, and granularity requirements. Here are a few examples.
```

File: website/blog/2024-11-04-test-smarter-not-harder.md [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+
+Get a pen and paper (or a google doc) and join us in constructing your own testing framework.
+
+## Identifying data quality issues in your pipeline
+
+Let’s start our framework by *identifying* types of data quality issues.
+
+In our daily work with customers, we find that data quality issues tend to fall into one of three broad buckets: *data hygiene, business-focused anomalies,* and *stats-focused anomalies.* Read the bucket descriptions below, and list 2-3 data quality concerns in your own business context that fall into each bucket.
+
+### Bucket One: Data Hygiene
```
Comment (joellabes) [inline website/blog/2024-11-04-test-smarter-not-harder.md]:
```suggestion
### Bucket 1: Data hygiene
```

File: website/blog/2024-11-04-test-smarter-not-harder.md [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+
+*Data hygiene* issues **are concerns you address in your [staging layer.](https://docs.getdbt.com/best-practices/how-we-structure/2-staging) Hygienic data meets your expectations around formatting, completeness, and granularity requirements. Here are a few examples.
+
+- *Granularity:* primary keys are unique and not null. Duplicates throw off calculations!
+- *Completeness:* columns that should always contain text, *do.* Incomplete data is less useful!
+- *Formatting:* email addresses always have a valid domain. Incorrect emails may affect marketing outreach!
+
+### Bucket Two: Business-focused Anomalies
```
Comment (joellabes) [inline website/blog/2024-11-04-test-smarter-not-harder.md]:
```suggestion
### Bucket 2: Business-focused anomalies
```

File: website/blog/2024-11-04-test-smarter-not-harder.md [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+
+Start at your business intelligence (BI) layer. Pick 1-3 dashboards or tables that you *know* are used frequently. List these 1-3 dashboards or tables. For each dashboard or table you have, identify 1-3 “expected” behaviors that your end-users rely on.  Here are a few examples to get you thinking:
+
+- Revenue numbers should not change by more than X% in Y amount of time. This could indicate fraud or OMS problems.
+- Monthly active users should not decline more than X% after the initial onboarding period. This might indicate user dissatisfaction, usability issues, or that users not finding a feature valuable.
+- Exam passing rates should stay above Y%.  A decline below that threshold may indicate recent content changes or technical issues are affecting understanding or accessibility.
+
+You should also consider what data issues you have had in the past! Look through recent data incidents and pick out 3 or 4 to guard against next time. These might be in a #data-questions channel or perhaps a DM from a stakeholder.
+
+### Bucket 3: Stats-focused Anomalies
```
Comment (joellabes) [inline website/blog/2024-11-04-test-smarter-not-harder.md]:
```suggestion
### Bucket 3: Stats-focused anomalies
```

File: website/blog/2024-11-04-test-smarter-not-harder.md [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+
+In our daily work with customers, we find that data quality issues tend to fall into one of three broad buckets: *data hygiene, business-focused anomalies,* and *stats-focused anomalies.* Read the bucket descriptions below, and list 2-3 data quality concerns in your own business context that fall into each bucket.
+
+### Bucket One: Data Hygiene
+
+*Data hygiene* issues **are concerns you address in your [staging layer.](https://docs.getdbt.com/best-practices/how-we-structure/2-staging) Hygienic data meets your expectations around formatting, completeness, and granularity requirements. Here are a few examples.
+
+- *Granularity:* primary keys are unique and not null. Duplicates throw off calculations!
+- *Completeness:* columns that should always contain text, *do.* Incomplete data is less useful!
+- *Formatting:* email addresses always have a valid domain. Incorrect emails may affect marketing outreach!
```
Comment (joellabes) [inline website/blog/2024-11-04-test-smarter-not-harder.md]: The examples here are good, I'm not convinced they need to end in !s. I also wonder if there's anything better than "less useful" for completeness?

File: website/blog/2024-11-04-test-smarter-not-harder.md (hunk 6 above)
Comment (joellabes) [reply website/blog/2024-11-04-test-smarter-not-harder.md]: I don't mind if you do One, Two, Three or 1, 2, 3 but you can't do One, Two, 3 😂

File: website/blog/2024-11-04-test-smarter-not-harder.md [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+- Monthly active users should not decline more than X% after the initial onboarding period. This might indicate user dissatisfaction, usability issues, or that users not finding a feature valuable.
+- Exam passing rates should stay above Y%.  A decline below that threshold may indicate recent content changes or technical issues are affecting understanding or accessibility.
+
+You should also consider what data issues you have had in the past! Look through recent data incidents and pick out 3 or 4 to guard against next time. These might be in a #data-questions channel or perhaps a DM from a stakeholder.
+
+### Bucket 3: Stats-focused Anomalies
+
+*Stats-focused anomalies* are fluctuations that go against your expected volumes or metrics.  Some examples include:
+
+- Volume anomalies. This could be site traffic amounts that may indicate illicit behavior, or perhaps site traffic dropping one day then doubling the next, indicating that data were not loaded properly.
```
Comment (joellabes) [inline website/blog/2024-11-04-test-smarter-not-harder.md]: Today was the day I discovered you're a "data is plural" person. 

I have no comment on this. (It's fine, you can leave it)

File: website/blog/2024-11-04-test-smarter-not-harder.md [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+
+To prioritize your data quality concerns, think about real-life impact. A couple of guiding questions to consider are:
+
+- Are your numbers *customer-facing?* For example, maybe you work with temperature-tracking devices. Your customers rely on these devices to show them average temperatures on perishable goods like strawberries in-transit. What happens if the temperature of the strawberries reads as 300C when they know their refrigerated truck was working just fine? How is your brand perception impacted when the numbers are wrong?
+- Are your numbers *used to make financial decisions?* For example, is the marketing team relying on your numbers to choose how to spend campaign funds?
+- Are your numbers *executive-facing?* Will executives use these numbers to reallocate funds or shift priorities?
+
+We think these 3 categories above constitute high-impact, pipeline-failing events. We think these categories should be your top priorities. Of course, adjust priority order if your business context calls for it.
```
Comment (joellabes) [inline website/blog/2024-11-04-test-smarter-not-harder.md]:
```suggestion
We think these 3 categories above constitute high-impact, pipeline-failing events, and should be your top priorities. Of course, adjust priority order if your business context calls for it.
```

File: website/blog/2024-11-04-test-smarter-not-harder.md [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+    - Discuss outliers with sales ops team
+
+You now have written out a prioritized list of data quality concerns, as well as action steps to take when each concern surfaces. Next, consult [hub.getdbt.com](http://hub.getdbt.com) and find tests that address each of your highest priority concerns. [dbt-expectations](https://hub.getdbt.com/calogica/dbt_expectations/latest/) and [dbt_utils](https://hub.getdbt.com/dbt-labs/dbt_utils/latest/) are great places to start.
+
+The data tests you’ve marked as “errors” above should get error-level severity. Any concerns falling into that nice-to-know category should either *not get tested* or have their tests *set to warning.*
+
+Your data quality priorities list is a living reference document. We recommend linking it in your project’s README so that you can go back and edit it as your testing needs evolve. Additionally, developers in your project should have easy access to this document. Maintaining good data quality is everyone’s responsibility!
+
+As you try these ideas out, come to the dbt community Slack and let us know what works and what doesn’t. Data is a community of practice, and we are eager to hear what comes out of yours.
```
Comment (joellabes) [inline website/blog/2024-11-04-test-smarter-not-harder.md]:
```suggestion
As you try these ideas out, come to the dbt Community Slack and let us know what works and what doesn’t. Data is a community of practice, and we are eager to hear what comes out of yours.
```

**Review by joellabes — CHANGES_REQUESTED** [review]
this rocks so hard. I'll be in DX offsite the next couple of days, but am happy for someone on @dbt-labs/product-docs to dismiss this review/approve once the last bits are mopped up!

File: website/blog/2024-11-04-test-smarter-not-harder.md (hunk 6 above)
Comment (faithebear) [reply website/blog/2024-11-04-test-smarter-not-harder.md]: dammit i really thought i had my numbers consistent 🫠  RIP!

File: website/blog/2024-11-04-test-smarter-not-harder.md (hunk 7 above)
Comment (faithebear) [reply website/blog/2024-11-04-test-smarter-not-harder.md]: changed this in a fresh commit & expanded a bit on incomplete data usually needing to get thrown out & reducing overall analytical power

File: website/blog/2024-11-04-test-smarter-not-harder.md (hunk 8 above)
Comment (faithebear) [reply website/blog/2024-11-04-test-smarter-not-harder.md]: clarified this a bit but i am a "data might be singular or plural depending on context" kind of person xD

**Pull request merged**